### DISCUSSION TITLE
Why does some text disappear when using a quoted command substitution in this expression?

### DISCUSSION THREAD

**Discussion by lacygoill** [fish-shell/fish-shell#8849] [Q&A] [answered]
Consider this snippet:
```fish
set x a.b.c
echo "$x$(printf d.e.f)"
```
It outputs this:

    .e.f

But I don't understand the result.  I would have expected this instead:

    a.b.cd.e.f

Just like in any of these similar snippets:
```fish
set x a.b.c
echo $x"$(printf d.e.f)"
```
    a.b.cd.e.f
```fish
set x a.b.c
echo "$x"$(printf d.e.f)
```
    a.b.cd.e.f
```fish
set x a.b.c
echo $x$(printf d.e.f)
```
    a.b.cd.e.f
```fish
set x a.b.c
set y "$(printf d.e.f)"
echo "$x$y"
```
    a.b.cd.e.f

The only syntax that I know of which might explain why some text disappears, is the cartesian product documented at `man fish-language`, section `PARAMETER EXPANSION`, subsection `Combining lists (Cartesian Product)`.

For example:
```fish
set x a b c
set --erase y
echo $x$y
```
But this requires one of the operands to evaluate to an empty list.  And in the original snippet, none of the operands evaluates to an empty list:
```fish
set x a.b.c
set y $(printf d.e.f)
set --show x
set --show y
```
    $x: set in global scope, unexported, with 1 elements
    $x[1]: |a.b.c|
    $y: set in global scope, unexported, with 1 elements
    $y[1]: |d.e.f|

Adding quotes around the command substitution in the `y` assignment does not change the results.

Where can I learn more about the syntax which explains why `a.b.cd` disappears?

**Comment by krobelus** [comment, marked as answer]
Oof that's embarassing. The behavior is wrong, we accidentally add the results of the command substitution to the variable name. Fixed by 1b668f567, thanks for reporting
